### PR TITLE
Minor revisions to BayesianTools RTM inversion

### PR DESCRIPTION
**Pull request by ashiklom**
## Description
- Add `max_iter` option to `invert_bt`.
- Minor bugfix in read settings.
- Faster tests (and test reading inversion options).

### CODE REVIEW

File: models/ed/R/write.configs.ed.R [hunk 1]
```diff
@@ -329,25 +329,25 @@ remove.config.ED2 <- function(main.outdir = settings$outdir, settings) {
...
   if (file.exists(system.file(histfile, package = "PEcAn.ED2"))) {
-    PEcAn.utils::logger.info(paste0("--- Using ED2 History File: ", "data/history.r", settings$model$revision, ".csv"))
-    edhistory <- read.csv2(system.file(histfile, package = "PEcAn.ED2"), sep = ";", 
+    PEcAn.utils::logger.debug(paste0("--- Using ED2 History File: ", "data/history.r", settings$model$revision, ".csv"))
+    edhistory <- read.csv2(system.file(histfile, package = "PEcAn.ED2"), sep = ";",
                            stringsAsFactors = FALSE, dec = ".")
   } else {
-    PEcAn.utils::logger.info("--- Using Generic ED2 History File: data/history.csv")
-    edhistory <- read.csv2(system.file("data/history.csv", package = "PEcAn.ED2"), sep = ";", 
+    PEcAn.utils::logger.debug("--- Using Generic ED2 History File: data/history.csv")
```
Comment (serbinsh) [inline models/ed/R/write.configs.ed.R]: I thought it was now PEcAn.base::logger.debug?

File: modules/rtm/R/bayestools.R [hunk 2]
```diff
@@ -67,98 +71,138 @@ prospect_bt_prior <- function(version, custom_prior = list()) {
...
                              init = list(iterations = 10000),
                              loop = list(iterations = 2000),
-                             other = list(sampler = 'DEzs', 
+                             other = list(sampler = 'DEzs',
                                           use_mpsrf = FALSE,
-                                          min_samp = 1000))
+                                          min_samp = 1000,
+                                          max_iter = 1e6,
+                                          lag.max = NULL,
+                                          save_progress = NULL))
```
Comment (serbinsh) [inline modules/rtm/R/bayestools.R]: So to be clear, to save progress I would do

```
invert_bt(observed, model, prior, custom_settings = list(save_progress=/some/directory/filename.rds))
```

Is that about right?

Comment (serbinsh) [inline modules/rtm/R/bayestools.R]: Sorry

```
invert_bt(observed, model, prior, custom_settings = list(other=list(save_progress=/some/directory/filename.rds)))
```

Comment (ashiklom) [inline modules/rtm/R/bayestools.R]: Not quite. `invert_bt(..., custom_settings = list(other = list(save_progress = "/path/to/file.rds")))`.

File: modules/rtm/R/bayestools.R [hunk 3]
```diff
@@ -67,98 +71,138 @@ prospect_bt_prior <- function(version, custom_prior = list()) {
...
+        stop_iter <- 10000
+        warning('init_settings$iterations is not set. Using ', stop_iter, '.')
+    }
+    message('Running initial ', stop_iter, ' iterations.')
+    samples <- BayesianTools::runMCMC(bayesianSetup = setup,
+                                      sampler = settings[['other']][['sampler']],
                                       settings = init_settings)
-    converged <- bt_check_convergence(samples = samples, use_mpsrf = settings[['other']][['use_mpsrf']])
+    if (!is.null(save_progress)) {
+        saveRDS(object = samples, file = save_progress)
```
Comment (serbinsh) [inline modules/rtm/R/bayestools.R]: Because here is looks like save_progress from above needs to be a file path

**Pull request merged**